### ISSUE TITLE
Remove some usage of memcache

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#18291]
As of today, here is the full usage of memcached in addons-server:

- `amo/celery.py`: cache start time for each task in order to compute total task time when it ends
- `amo/utils.py`: a cache namespacing utility, that uses an intermediary cache key to be able to clear an entire key space (e.g. be able to clear all cache keys for a specific feature). Only used in reviewer tools score cache, so if that
  goes we could remove it entirely.
- `lib/cache.py`: `memoize` and friends, to be removed, see https://github.com/mozilla/addons-server/issues/18085
- `ratings/tasks.py`: cache all rating/total ratings aggregates for all valid add-ons for 60 hours. Not sure how useful that is, yes the task is called often but the query is super fast. Could be ran on a replica though, and should probably be limited to
  public add-ons.
- `devhub/tasks.py`: cache the validation task id for a given file upload, so that we don't validate the same thing twice. We should investigate whether that's really useful, cause it should not happen - this was from a time when we could have unvalidated uploads  laying around and reviewers could manually trigger validation (https://bugzilla.mozilla.org/show_bug.cgi?id=1191603)
- `users/models.py`: cache denied names for default duration (500 seconds)
- `applications/views.py`: cache list of applications versions for default duration
- `reviewers/models.py` and `reviewers/views.py`:
  - cache addon id being reviewed to "lock" it for a short period
  - cache a bunch of reviewer score related things (user score, recent points awarded, breakdown for a user, breakdown for a user since a certain date, leaderboard). https://github.com/mozilla/addons-server/issues/18049 might remove that.
- all throttling code
- views cached using `@cache_page` - see https://github.com/mozilla/addons-server/issues/18083 which will remove them (we should rely on the CDN for that)

Investigate what is still useful and what could be removed.

**Comment by eviljeff** [comment]
stumbled upon this old issue - is any of it still valid @diox ?

**Comment by diox** [comment]
I think so. We did a few but a bunch of these remain, need to investigate those bullet points.

**Comment by diox** [comment]
I went through the list to clarify what needs to be done.

**Comment by eviljeff** [comment]
> users/models.py: cache denied names for default duration (500 seconds). Investigate usefulness

I doubt there are frequent enough display name changes to justify this. (nor collection names).

**Comment by diox** [comment]
Agreed. Also, we don't have to fetch everything all the time, we can do a ["reverse LIKE" query](https://stackoverflow.com/questions/15465858/django-reverse-to-contains-icontains/57098265#57098265).

**Comment by KevinMind** [comment]
Old Jira Ticket: https://mozilla-hub.atlassian.net/browse/ADDSRV-65
